modules/spark/src/main/scala/org/apache/ignite/spark/impl/package.scala

```diff
@@ -124,10 +130,63 @@ package object impl {
     def sqlTableInfo[K, V](ignite: Ignite, tabName: String,
         schemaName: Option[String]): Option[(CacheConfiguration[K, V], QueryEntity)] =
         cachesForSchema[K,V](ignite, schemaName)
-            .map(ccfg ⇒ ccfg.getQueryEntities.find(_.getTableName.equalsIgnoreCase(tabName)).map(qe ⇒ (ccfg, qe)))
+            .map(ccfg ⇒ refreshQueryEntity(ignite, ccfg,
+                ccfg.getQueryEntities.find(_.getTableName.equalsIgnoreCase(tabName))).map(qe ⇒ (ccfg, qe)))
             .find(_.isDefined)
             .flatten
 
+    /**
+      * @param ignite Ignite instance.
+      * @param ccfg Cache Configuration.
+      * @param qe Query Entity.
+      * @tparam K Key class.
+      * @tparam V Value class.
+      * @return QueryEntity for a given table.
+      */
+    def refreshQueryEntity[K, V](ignite: Ignite, ccfg : CacheConfiguration[K, V], qe: Option[QueryEntity]): Option[QueryEntity] =
+        if (qe.isDefined) Some(qe.get.setFields(refreshFields(ignite, ccfg, qe.get.getTableName)))
+        else qe
+
+    /**
+      * @param ignite Ignite instance.
+      * @param ccfg Cache Configuration.
+      * @param tabName Table name.
+      * @tparam K Key class.
+      * @tparam V Value class.
+      * @return HashMap consists of fields and fields type for a given table.
+      */
+    def refreshFields[K, V](ignite: Ignite, ccfg : CacheConfiguration[K, V], tabName: String): util.LinkedHashMap[String, String] = {
+        val connection = connect(ignite)
```

Comment:
Seems, we can do this without jdbc connection, just using internal Ignite API.
Please, take a look at 

`org.apache.ignite.internal.processors.odbc.jdbc.JdbcRequestHandler#getColumnsMeta`

```
            for (String cacheName : ctx.cache().publicCacheNames()) {
                for (GridQueryTypeDescriptor table : ctx.query().types(cacheName)) {
                    if (!matches(table.schemaName(), req.schemaName()))
                        continue;

                    if (!matches(table.tableName(), req.tableName()))
                        continue;
        //Process column here
```